common/persistence/elasticsearch/esVisibilityStore.go

```diff
@@ -217,7 +227,12 @@ func (v *esVisibilityStore) ListOpenWorkflowExecutionsByType(
 		}
 	}
 
-	return v.getListWorkflowExecutionsResponse(searchResult.Hits, token, definition.StartTime, request.PageSize)
+	isRecordValid := func(rec *p.VisibilityWorkflowExecutionInfo) bool {
```

Comment:
Looks like it's the same function as L181. Declare it as a normal function instead of a lambda function so it can be reused?

Reply:
or declare it in getListWorkflowExecutionsResponse() and pass isOpen to that function?